integration-cli/docker_cli_tag_test.go

```diff
@@ -86,30 +86,6 @@ func (s *DockerSuite) TestTagExistedNameWithoutForce(c *check.C) {
 	dockerCmd(c, "tag", "busybox:latest", "busybox:test")
 }
 
-func (s *DockerSuite) TestTagWithPrefixHyphen(c *check.C) {
-	// Don't attempt to pull on Windows as not in hub. It's installed
-	// as an image through .ensure-frozen-images-windows
-	if daemonPlatform != "windows" {
-		if err := pullImageIfNotExist("busybox:latest"); err != nil {
-			c.Fatal("couldn't find the busybox:latest image locally and failed to pull it")
-		}
-	}
-	// test repository name begin with '-'
-	out, _, err := dockerCmdWithError("tag", "busybox:latest", "-busybox:test")
-	c.Assert(err, checker.NotNil, check.Commentf(out))
-	c.Assert(out, checker.Contains, "Error parsing reference", check.Commentf("tag a name begin with '-' should failed"))
-
```

Comment:
I don't think we should remove these tests. I think we can fix the behaviour.

If you check out `run`, I use `flags.SetInterspersed(false)` in the `NewRunCommand()`. This will making it stop looking for flags once it finds the first non-flag arg.

Once that' s done we should get the `-busybox:test` arg as an arg, instead of failing flag validation, which should let us get the old error message out.